docs/developers_notes/06-glm.md

```diff
@@ -0,0 +1,91 @@
+# The `glm` Module
+
+## Introduction
+
+
+
+Generalized Linear Models (GLM) provide a flexible framework for modeling a variety of data types while establishing a relationship between multiple predictors and a response variable. A GLM extends the traditional linear regression by allowing for response variables that have error distribution models other than a normal distribution, such as binomial or Poisson distributions.
+
+The `nemos.glm` module currently  offers implementations of two GLM classes:
+
+1. **`GLM`:** A direct implementation of a feedforward GLM.
+2. **`PopulationGLM`:** An implementation of a GLM for fitting a populaiton of neuron in a vectorized manner. This class inherits from `GLM` and redefines the `fit` and `_predict` to fit the model and predict the firing rate.
+
+Our design aligns with the `scikit-learn` API, facilitating seamless integration of our GLM classes with the well-established `scikit-learn` pipeline and its cross-validation tools.
+
+The classes provided here are modular by design offering a standard foundation for any GLM variant. 
+
+Instantiating a specific GLM simply requires providing an observation model (Gamma, Poisson, etc.), a regularization strategies (Ridge, Lasso, etc.) and an optimization scheme during initialization. This is done using the [`nemos.observation_models.Observations`](../03-observation_models/#the-abstract-class-observations), [`nemos.regularizer.Regularizer`](../04-regularizer/#the-abstract-class-regularizer) objects as well as the compatible `jaxopt` solvers, respectively.
+
+
+<figure markdown>
+    <img src="../GLM_scheme.jpg"/>
+    <figcaption>Schematic of the module interactions.</figcaption>
+</figure>
+
```

Comment:
while I love a good schematic, this needs a key: what do the arrows mean? the blue  diamonds?

Reply:
also it's out of date: lacks `GammaObservations`. Keeping this up-to-date could be annoying, so maybe either say "this shows some of the options" or include the script to generate it and have it run as part of the docs buidl

Reply:
updated!